### PR TITLE
updating to linkerd 0.8.3, api auth support

### PR DESCRIPTION
**Pull request by siggy**
changes:
- added marathon api auth support, for DC/OS Enterprise
- added marathon host, port, and uri prefix options
- updated linkerd from 0.8.1 to 0.8.3

### CODE REVIEW

File: repo/packages/L/linkerd/8/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "packagingVersion": "3.0",
+  "name": "linkerd",
+  "version": "0.8.3-0.1",
+  "scm": "https://github.com/BuoyantIO/linkerd/",
+  "description": "A dynamic linker for microservices. Install, set your http proxy to $HOST:4140, and connect to your services via app name. linkerd will handle Marathon service discovery for you.",
+  "maintainer": "hello@buoyant.io",
+  "website": "https://linkerd.io",
+  "tags": ["linkerd", "namerd", "buoyant", "loadbalancer", "service-discovery", "proxy", "microservices", "rpc"],
+  "preInstallNotes": "We recommend a minimum of 0.25 CPUs and 256 MB of RAM available for linkerd. We also recommend running linkerd on every host. To ensure this, set instances to the number of nodes in your cluster.",
```
Comment (ryadav88) [inline repo/packages/L/linkerd/8/package.json]: Add a linkerd's `examples` link at the end of `preInstallNotes`. 
Something like: `Installation Documentation: https://github.com/dcos/examples/tree/master/1.8/linkerd`

File: repo/packages/L/linkerd/8/config.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        },
+        "marathon-uri-prefix": {
+          "default": "",
+          "description": "Marathon API prefix. Typically left blank, or '/marathon'.",
+          "type": "string"
+        },
+        "secret_name": {
+          "default": "",
+          "description": "Name of the Secret Store credentials to use for DC/OS service authentication. This should be left empty unless service authentication is needed. For more information see https://docs.mesosphere.com/1.8/administration/id-and-access-mgt/service-auth/custom-service-auth/",
+          "type": "string"
```
Comment (ryadav88) [inline repo/packages/L/linkerd/8/config.json]: please update the `dcos/examples` to describe these new properties and also how the installation method changes. https://github.com/dcos/examples/tree/master/1.8/linkerd

File: repo/packages/L/linkerd/8/config.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        },
+        "marathon-uri-prefix": {
+          "default": "",
+          "description": "Marathon API prefix. Typically left blank, or '/marathon'.",
+          "type": "string"
+        },
+        "secret_name": {
+          "default": "",
+          "description": "Name of the Secret Store credentials to use for DC/OS service authentication. This should be left empty unless service authentication is needed. For more information see https://docs.mesosphere.com/1.8/administration/id-and-access-mgt/service-auth/custom-service-auth/",
```
Comment (ryadav88) [inline repo/packages/L/linkerd/8/config.json]: Add `Enterprise Only` at the start of `description`

File: repo/packages/L/linkerd/8/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "packagingVersion": "3.0",
+  "name": "linkerd",
+  "version": "0.8.3-0.1",
```
Comment (ryadav88) [inline repo/packages/L/linkerd/8/package.json]: How are you managing the `version` here? I see that previous Universe version was `0.8.1-0.3`
The format should be linkerdVersion-packagingVersion.

**Comment by ryadav88** [comment]
@siggy Does 0.8.3 bring some new features or has fixes from the previous ones? Can I see the release notes somewhere or can you describe them in a comment here? 

I'm asking so I can make a note in next month's Universe blog post. icymi this was November's post: https://dcos.io/blog/2016/new-dc-os-packages-you-can-use-november-2016/index.html

File: repo/packages/L/linkerd/8/config.json (hunk 3 above)
Comment (siggy) [inline repo/packages/L/linkerd/8/config.json]: Is it possible that Marathon API Auth can be enabled for community edition? If not, happy to mark this `Enterprise Only`.

**Comment by siggy** [comment]
@ryadav88 Full linkerd release notes are at: https://github.com/BuoyantIO/linkerd/releases. However, that does not necessarily capture the full changes, as their are features/options we can enable that are specific to this universe package. I'll document more complete changes when I update this PR.

File: repo/packages/L/linkerd/8/package.json (hunk 4 above)
Comment (ryadav88) [inline repo/packages/L/linkerd/8/package.json]: cool. atm we're letting package maintainers decide how they want to manage it. I just want to understand. You may keep as it is.

**Comment by ryadav88** [comment]
> @ryadav88 Full linkerd release notes are at: https://github.com/BuoyantIO/linkerd/releases. However, that does not necessarily capture the full changes, as their are features/options we can enable that are specific to this universe package. I'll document more complete changes when I update this PR.

thanks. looking forward to it.

File: repo/packages/L/linkerd/8/config.json (hunk 2 above)
Comment (siggy) [inline repo/packages/L/linkerd/8/config.json]: https://github.com/dcos/examples/pull/28

**Comment by siggy** [comment]
@ryadav88 changes since last linkerd dc/os universe package:

```
- added Marathon auth support for DC/OS Enterprise
- added Marathon host, port, and uriPrefix options
- updated linkerd from 0.8.1 to 0.8.3
  - show transformers in the delegate UI
  - improve HTTP server behavior with short-lived connections
  - improve retry metrics to include a total counter of all retry requests
  - fix crash when viewing the dtab playground
```

**Pull request merged**